.lintr

```diff
@@ -1,45 +1,32 @@
-linters: linters_with_defaults(
-    any_duplicated_linter(),
-    any_is_na_linter(),
-    backport_linter("oldrel-4", except = c("R_user_dir", "str2lang", "str2expression", "deparse1", "...names")),
-    consecutive_assertion_linter(),
-    expect_comparison_linter(),
-    expect_identical_linter(),
-    expect_length_linter(),
-    expect_named_linter(),
-    expect_not_linter(),
-    expect_null_linter(),
-    expect_s3_class_linter(),
-    expect_s4_class_linter(),
-    expect_true_false_linter(),
-    expect_type_linter(),
-    fixed_regex_linter(),
-    for_loop_index_linter(),
-    if_not_else_linter(),
-    implicit_assignment_linter(),
-    implicit_integer_linter(),
-    keyword_quote_linter(),
-    lengths_linter(),
+linters: all_linters(
+    backport_linter("3.6.0", except = c("R_user_dir", "deparse1", "...names")),
     line_length_linter(120L),
-    missing_argument_linter(),
-    nested_ifelse_linter(),
-    numeric_leading_zero_linter(),
-    outer_negation_linter(),
-    paste_linter(),
-    redundant_equals_linter(),
-    redundant_ifelse_linter(),
-    sort_linter(),
-    sprintf_linter(),
-    strings_as_factors_linter(),
-    undesirable_function_linter(c(Sys.setenv = NA_character_, mapply = NA_character_, structure = NA_character_)),
-    unnecessary_nested_if_linter(),
-    unnecessary_lambda_linter(),
+    object_overwrite_linter(
+      allow_names = c("end", "line", "lines", "new", "offset", "pipe", "start", "symbols")
```

Comment:
Where are the lints corresponding to the exceptions? Maybe we can nolint some sections instead?

Reply:
They're all over:

```r
                       filename line_number            object
 1:  R/get_source_expressions.R         638        stats::end
 2:  R/get_source_expressions.R         642        stats::end
 3:              R/path_utils.R         147        stats::end
 4:  R/get_source_expressions.R         205       stats::line
 5:  R/get_source_expressions.R         236       stats::line
 6:  R/get_source_expressions.R         239       stats::line
 7:                    R/lint.R         650       stats::line
 8:  R/get_source_expressions.R         564   graphics::lines
 9:                    R/lint.R          43   graphics::lines
10:                    R/lint.R         362   graphics::lines
11:                   R/utils.R         182   graphics::lines
12:      R/xml_nodes_to_lints.R          69   graphics::lines
13:      R/xml_nodes_to_lints.R          70   graphics::lines
14: tests/testthat/test-cache.R           7   graphics::lines
15:                    R/lint.R         338      methods::new
16:                    R/lint.R         341      methods::new
17:  R/get_source_expressions.R         587     stats::offset
18:    R/one_call_pipe_linter.R          73        base::pipe
19:        R/pipe_call_linter.R          34        base::pipe
20:  R/get_source_expressions.R         633      stats::start
21:              R/path_utils.R         146      stats::start
22:        R/namespace_linter.R          97 graphics::symbols
23:     R/object_usage_linter.R         138 graphics::symbols
24:        R/shared_constants.R         294 graphics::symbols
                       filename line_number            object
```

Reply:
I took another look and renamed `start`, `end`, and `#nolint`'d `offset`.

I think the other names are good to ignore everywhere. `line`/`lines`/`pipe`/`symbols` all will continue to come up reasonably often for us.

Reply:
OK, also renamed the arguments for internal function `lintr_deprecated` to avoid `new`.